### PR TITLE
Nuke: extract baked review videos presets

### PR DESCRIPTION
**Pull request by jakubjezek001**
Client had requested two different review video looks - with baked viewer input process and without. To be able to do this we have to refactor the way nuke is baking mov file and enhance extract plugin with presets. Those allowing to set baking parameters per task-type and family filters. 

This PR includes:
- [x] Nuke Extract review baked mov with presets
- [x] Nuke ImageIO settigs had been enriched with  `project_anatomy/imageio/nuke/baking/viewerProcess` as the baking is usually needed to be in `rec709` rather then in sRGB 
image

- [x] Global extract review settings added burnin profile linking
- [x] Global extract burnin accepting any linked profiles defined in representation data. 
- [x] Global extract burnin script had been fixed to be able to read source timecode from its metadata

# Known issues caused by this PR:
When representation is created the name is dynamically joined upstream from ExtractReviewDataMov down to burnins. But Load clip in Nuke is not able to recognize those representation names as it is working only with extract review profile names. 

# Testing:
1. first of all you had to open settings and define couple different profiles `project_settings/nuke/publish/ExtractReviewDataMov/outputs`. For example 
image
2. Create one more Burnin profile and change a little bit burnins so they are different then the default.  
3.  Create one more additional Extract review profile's output (could be copy of h264) but name it differently.  Ideally use following mxf settings 
image

4. In new output you had just created go to `Link to a burnin by name` and add the name of the burnin profile you had just crated in 2. 
image
  
5. Then open in nuke some of yours testing scenes
6. Create Render node and set it to plublish with Local rendering  and hit publish.
7. At the end you go to plublished Render folder and see all those review videos are there with the burnins you had set.

### CODE REVIEW

**Comment by mkolar** [comment]
Task linked: [OP-1933 [TGN] setting some defaults](https://app.clickup.com/t/6658547/OP-1933)

**Comment by iLLiCiTiT** [comment]
It seems that representation must have set burnins links otherwise no burnins are created so it will break all current burnins creation, or?

**Comment by jakubjezek001** [comment]
This issue https://github.com/pypeclub/OpenPype/issues/2252 would support the Extract review/burnin in better filtering. So for example if a particular tag was added into ExtractReviewDataMov profile's Tag ten we could more precisely target for the review and burnin outputs.

File: openpype/plugins/publish/extract_burnin.py [hunk 1]
```diff
@@ -126,18 +129,44 @@ def main_process(self, instance):
...
+            repre_linked_burnins = {
+                name: output for name, output in burnin_defs.items()
+                if name in repre_burnin_links
+            }
+            self.log.debug("repre_linked_burnins: {}".format(
+                repre_linked_burnins))
+
+            # if any match then replace burnin defs and follow tag filtering
+            _burnin_defs = copy.deepcopy(burnin_defs)
+            if repre_linked_burnins:
```
Comment (jakubjezek001) [inline openpype/plugins/publish/extract_burnin.py:164]: https://github.com/pypeclub/OpenPype/pull/2248#issuecomment-969102543

@iLLiCiTiT I believe this is adding backward compatibility and securing that if no link had been captured then it jumps to old way of processing burn-in outputs.

**Comment by jakubjezek001** [comment]
This PR also holds content of https://github.com/pypeclub/OpenPype/pull/2247

**Comment by jakubjezek001** [comment]
This issue had occurred during tests and it is something which would be great to address. https://github.com/pypeclub/OpenPype/issues/2253

File: openpype/lib/plugin_tools.py [hunk 2]
```diff
@@ -531,12 +531,20 @@ def should_decompress(file_url):
                 and we can decompress (oiiotool supported)
     """
     if oiio_supported():
-        output = run_subprocess([
-            get_oiio_tools_path(),
-            "--info", "-v", file_url])
-        return "compression: \"dwaa\"" in output or \
-            "compression: \"dwab\"" in output
-
+        try:
```
Comment (jakubjezek001) [inline openpype/lib/plugin_tools.py:534]: @iLLiCiTiT or @antirotor I have add this here but still not sure if this is the correct way to do it.

File: openpype/lib/plugin_tools.py [hunk 3]
```diff
@@ -531,12 +531,20 @@ def should_decompress(file_url):
...
+            return "compression: \"dwaa\"" in output or \
+                "compression: \"dwab\"" in output
+        except RuntimeError as _E:
+            _name, ext = os.path.splitext(file_url)
+            if ext in [".mxf"]:
+                # TODO: should't the list of allowed extensions be
+                # taken from an OIIO variable of supported formats
+                return False
+            else:
+                raise RuntimeError(_E)
```
Comment (iLLiCiTiT) [inline openpype/lib/plugin_tools.py:547]:
```suggestion
        except RuntimeError:
            _name, ext = os.path.splitext(file_url)
            if ext not in [".mxf"]:
                # Reraise exception
                raise
               
            # TODO: should't the list of allowed extensions be
            # taken from an OIIO variable of supported formats
            return False
```

File: openpype/plugins/publish/extract_burnin.py [hunk 4]
```diff
@@ -281,14 +307,16 @@ def main_process(self, instance):
             # NOTE we maybe can keep source representation if necessary
             instance.data["representations"].remove(repre)
 
-            # Delete input files
```
Comment (jakubjezek001) [inline openpype/plugins/publish/extract_burnin.py:284]: this has to be returned back, there is certainly duplicity in representations if this is not letting rest of the process to go on

Comment (jakubjezek001) [reply openpype/plugins/publish/extract_burnin.py:284]: altaught this had been reveresed and it is reflected in the code it is not displayed here in the changes so be aware that this is not showing correct state..

Comment (jakubjezek001) [reply openpype/plugins/publish/extract_burnin.py:284]: this is the commit reversing it.. [`dd27953` (#2248)](https://github.com/pypeclub/OpenPype/pull/2248/commits/dd27953068efe35579ae77fb3401e539ffaef041)

File: openpype/hosts/nuke/plugins/publish/extract_review_data_mov.py [hunk 5]
```diff
@@ -45,28 +45,65 @@ def process(self, instance):
...
-            exporter = pnlib.ExporterReviewMov(
-                self, instance)
+            for o_name, o_data in self.outputs.items():
+                f_families = o_data["filter"]["families"]
+                f_task_types = o_data["filter"]["task_types"]
 
-            if "render.farm" in families:
-                instance.data["families"].remove("review")
-                data = exporter.generate_mov(farm=True)
+                test_families = any([
```
Comment (iLLiCiTiT) [inline openpype/hosts/nuke/plugins/publish/extract_review_data_mov.py:56]: This is super confusing for me. I've looked at the code for 5 minutes and I had to rewrite it to understand what are you trying to do...

Comment (jakubjezek001) [reply openpype/hosts/nuke/plugins/publish/extract_review_data_mov.py:56]: eh.. yeah you are right.. I will add comments to explain the process

Comment (jakubjezek001) [reply openpype/hosts/nuke/plugins/publish/extract_review_data_mov.py:57]: I hope it makes more sence now from comments

**Review by 64qam — APPROVED** [review]
Looks good to me

**Pull request merged**